src/Ui/UiRequest.py

```diff
@@ -40,11 +41,17 @@ def __init__(self, server, get, env, start_response):
         self.start_response = start_response  # Start response function
         self.user = None
 
+        self.FS_ENCODING = sys.getfilesystemencoding()
+
     # Call the request handler function base on path
     def route(self, path):
         if config.ui_restrict and self.env['REMOTE_ADDR'] not in config.ui_restrict:  # Restict Ui access by ip
             return self.error403(details=False)
 
+        self.log.debug("-- path repr: " + repr(path))
+        path = path.decode('utf-8')
+        self.log.debug("-- path decoded repr: " + repr(path))
+
```

Comment:
If you are annoyed by encoding issues of Python 2, try using `repr(...)` on strings when printing out debug log.